### PR TITLE
Update stable releases

### PR DESCRIPTION
**Pull request by asauber**
Update Readme with latest stable releases.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -80,7 +80,7 @@ corresponding image pull tags and their release notes where applicable:
...
 | `main <https://github.com/cilium/cilium/commits/main>`__                             | daily      | ``quay.io/cilium/cilium-ci:latest``          | N/A                                                                                      |
 +--------------------------------------------------------------------------------------+------------+----------------------------------------------+------------------------------------------------------------------------------------------+
-| `v1.14.0-rc.1 <https://github.com/cilium/cilium/commits/v1.14.0-rc.1>`__             | 2023-07-17 | ``quay.io/cilium/cilium:v1.14.0-rc.1``       | `Release Candidate Notes <https://github.com/cilium/cilium/releases/tag/v1.14.0-rc.1>`__ |
+| `v1.14.1-rc.1 <https://github.com/cilium/cilium/commits/v1.14.1-rc.1>`__             | 2023-07-17 | ``quay.io/cilium/cilium:v1.14.1-rc.1``       | `Release Candidate Notes <https://github.com/cilium/cilium/releases/tag/v1.14.1-rc.1>`__ |
```
Comment (joestringer) [inline README.rst]: I guess this got caught up in some `sed` expression, but there's no such release:

```suggestion
| `v1.14.0-rc.1 <https://github.com/cilium/cilium/commits/v1.14.1-rc.1>`__             | 2023-07-17 | ``quay.io/cilium/cilium:v1.14.1-rc.1``       | `Release Candidate Notes <https://github.com/cilium/cilium/releases/tag/v1.14.0-rc.1>`__ |
```

**Review by joestringer — APPROVED** [review]
Approve except for one minor discrepancy below.

File: stable.txt [hunk 2]
```diff
@@ -1 +1 @@
-v1.14.0
+v1.14.1
```
Comment (zacharysarah) [inline stable.txt:1]: I'm guessing @joestringer's [comment](https://github.com/cilium/cilium/pull/27637/files#r1302236122) ☝🏻 applies here too.

**Review by zacharysarah — APPROVED** [review]
LGTM for docs

File: stable.txt (hunk 2 above)
Comment (joestringer) [reply stable.txt:1]: This is correct. v1.14.1 is the latest release, it's the bit about v1.14.1-rc.1 which has no corresponding release.

File: README.rst (hunk 1 above)
Comment (asauber) [reply README.rst]: Now that we have 1.14.1, should we remove this section?

It was added for 1.14.0: https://github.com/cilium/cilium/commit/44756bad5b826e0ff13e9f31e1160f9f102ad396

Perhaps we replace the last paragraph and table with: "There is no current release candidate build."

Comment (joestringer) [reply README.rst]: This particular line will be updated in about a week when we do the first v1.15.0-pre.X release.

Comment (joestringer) [reply README.rst]: So feel free to either drop it or just ignore this line for now.

Comment (asauber) [reply README.rst]: Left it as is

**Pull request merged**